File: app/src/main/java/com/selfmate/mes/selfmate/Models/MoodLogDao.java
package com.selfmate.mes.selfmate.Models;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

/**
 * Created by vicki_mes on 11/16/2017.
 */


@Dao
public interface MoodLogDao {

    @Insert
    void insertMoodToLog(MoodLog moodLog);

    @Update
    void updateMoodToLog(MoodLog moodLog);

    @Query("Select * from moodlog")
    List<MoodLog> getMoogLog();
}

File: app/src/main/java/com/selfmate/mes/selfmate/models/MoodLogDao.kt
package com.selfmate.mes.selfmate.models

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query
import android.arch.persistence.room.Update

/**
 * Created by vicki_mes on 11/16/2017.
 */


@Dao
interface MoodLogDao {

    @get:Query("Select * from moodlog")
    val moogLog: LiveData<List<MoodLog>>

    @Insert
    fun insertMoodToLog(moodLog: MoodLog): Long

    @Update
    fun updateMoodToLog(moodLog: MoodLog): Int

    @Query("select * from moodlog ORDER BY time LIMIT  1")
    fun getLatestMood(): LiveData<MoodLog>
}
